tensorflow/core/kernels/mkl_fused_batch_norm_op.cc

```diff
@@ -396,60 +533,78 @@ class MklFusedBatchNormBwdPrimitive : public MklPrimitive {
   };
 
   void Setup(const MklBatchNormBwdParams& bwdParams) {
-    context_.flags = bwdParams.training ? use_scale_shift
-                                        : (use_scale_shift | use_global_stats);
+    context_.flags = bwdParams.training ? (int)BN_FLAGS::use_scale_shift
+                                        : ((int)BN_FLAGS::use_scale_shift |
+                                           (int)BN_FLAGS::use_global_stats);
 
-    // memory desc
+    // Memory descriptors.
+#ifndef ENABLE_MKL_DNN_V1
     auto src_md = memory::desc({bwdParams.src_dims}, MklDnnType<T>(),
                                bwdParams.src_format);
     auto diff_dst_md = memory::desc({bwdParams.diff_dst_dims}, MklDnnType<T>(),
                                     bwdParams.src_format);
+#else
+    auto src_md = memory::desc({bwdParams.src_dims}, MklDnnType<T>());
+    auto diff_dst_md = memory::desc({bwdParams.diff_dst_dims}, MklDnnType<T>());
+#endif
     auto variance_desc =
-        memory::desc({1, bwdParams.depth}, MklDnnType<U>(), memory::nc);
+        memory::desc({1, bwdParams.depth}, MklDnnType<U>(), MEMORY_FORMAT::nc);
     auto mean_desc =
-        memory::desc({1, bwdParams.depth}, MklDnnType<U>(), memory::format::nc);
+        memory::desc({1, bwdParams.depth}, MklDnnType<U>(), MEMORY_FORMAT::nc);
     auto weights_desc =
-        memory::desc({2, bwdParams.depth}, MklDnnType<U>(), memory::format::nc);
+        memory::desc({2, bwdParams.depth}, MklDnnType<U>(), MEMORY_FORMAT::nc);
     auto diff_weights_desc = weights_desc;
 
-    // fwd desc & primitive desc
+    // Forward batch-normalization descriptor and primitive descriptor.
+    auto bn_flags =
+        bwdParams.training
+            ? BN_FLAGS::use_scale_shift
+            : (BN_FLAGS::use_scale_shift | BN_FLAGS::use_global_stats);
```

Comment:
How is this different from `context_.flags`?

Reply:
Changed code to use context_.flags. Removed the redundant code